internal/federation/server.go

```diff
@@ -200,7 +201,9 @@ func (s *Server) Listen() (cancel func()) {
 		defer wg.Done()
 		err := s.srv.ListenAndServeTLS(s.certPath, s.keyPath)
 		if err != nil && err != http.ErrServerClosed {
-			s.t.Fatalf("ListenFederationServer: ListenAndServeTLS failed: %s", err)
+			s.t.Logf("ListenFederationServer: ListenAndServeTLS failed: %s", err)
+			// Note that running s.t.FailNow is not allowed in a separate goroutine
+			// TODO: Figure out a way to fail the test here
```

Comment:
We probably shouldn't fail the test in retrospect, as this error is from a cleanup process. If the server never listened in the first place (e.g conflicting port numbers) then this will result in `connection refused` errors in tests.

Reply:
Good point! I'll remove the TODO.